test/ex_webrtc/dtls_transport_test.exs

```diff
@@ -0,0 +1,132 @@
+defmodule ExWebRTC.DTLSTransportTest do
+  use ExUnit.Case, async: true
+
+  alias ExWebRTC.DTLSTransport
+
+  defmodule FakeICEAgent do
+    use GenServer
+
+    def start_link(_mode, config) do
+      GenServer.start_link(__MODULE__, config)
+    end
+
+    def send_data(ice_agent, data) do
+      GenServer.cast(ice_agent, {:send_data, data})
+    end
+
+    @impl true
+    def init(tester: tester), do: {:ok, tester}
+
+    @impl true
+    def handle_cast({:send_data, data}, tester) do
+      send(tester, {:fake_ice, data})
+      {:noreply, tester}
+    end
+  end
+
+  setup do
+    assert {:ok, dtls} = DTLSTransport.start_link([tester: self()], FakeICEAgent)
+
+    %{dtls: dtls}
+  end
+
+  test "forwards non-data ICE messages", %{dtls: dtls} do
+    message = "test message"
+
+    send_ice(dtls, message)
+    assert_receive {:ex_ice, _from, ^message}
+
+    send_ice(dtls, {:data, <<1, 2, 3>>})
+    refute_receive {:ex_ice, _from, _msg}
+  end
+
+  test "cannot send data when handshake not finished", %{dtls: dtls} do
+    DTLSTransport.send_data(dtls, <<1, 2, 3>>)
+
+    refute_receive {:fake_ice, _data}
+  end
+
+  test "cannot start dtls more than once", %{dtls: dtls} do
+    assert :ok = DTLSTransport.start_dtls(dtls, :passive)
+    assert {:error, :already_started} = DTLSTransport.start_dtls(dtls, :passive)
+  end
+
+  test "initiates DTLS handshake when in active mode", %{dtls: dtls} do
+    :ok = DTLSTransport.start_dtls(dtls, :active)
+
+    send_ice(dtls, :connected)
+
+    assert_receive {:fake_ice, packets}
+    assert is_binary(packets)
+  end
+
+  test "won't initiate DTLS handshake when in passive mode", %{dtls: dtls} do
+    :ok = DTLSTransport.start_dtls(dtls, :passive)
+
+    send_ice(dtls, :connected)
+
+    refute_receive({:fake_ice, _msg})
+  end
+
+  test "will retransmit after initiating handshake", %{dtls: dtls} do
+    :ok = DTLSTransport.start_dtls(dtls, :active)
+
+    send_ice(dtls, :connected)
+
+    assert_receive {:fake_ice, _packets}
+    assert_receive {:fake_ice, _retransmited}, 1200
```

Comment:
assert packets and retransmitted are the same

Reply:
But they are not the same tho.